docs/index.asciidoc

```diff
@@ -25,21 +25,62 @@ Stream events from files, normally by tailing them in a manner
+By default, each event is assumed to be one line
+and a line is taken to be the text before a newline character.
 If you would like to join multiple log lines into one event,
-you'll want to use the multiline codec or filter.
-
-The plugin aims to track changing files and emit new content as it's
-appended to each file. It's not well-suited for reading a file from
-beginning to end and storing all of it in a single event (not even
-with the multiline codec or filter).
+you'll want to use the multiline codec.
+The plugin loops between discovering new files and processing
+each discovered file. Discovered files have a lifecycle, they start off
+in the "watched" or "ignored" state. Other states in the lifecycle are:
+"active", "closed" and "unwatched"
+
+By default, a window of 4095 files is used to limit the number of file handles in use.
+The processing phase has a number of stages:
+
+. checks whether "closed" or "ignored" files have
+changed in size since last time and if so puts them in the "watched" state.
+. selects enough "watched" files to fill the available space in the window,
+these files are made "active".
+. the active files are opened and read, each file is read from the
+last known position to the end of current content (EOF) by default.
+
+In some cases it is useful to be able to control which files are read first, sorting,
+and whether files are read completely or banded/striped.
+Complete reading is *all of* file A then file B then file C and so on.
+Banded or striped reading is *some of* file A then file B then file C and so on looping around
+to file A again until all files are read. Banded reading is specified by changing
+<<plugins-{type}s-{plugin}-file_chunk_count>> and perhaps <<plugins-{type}s-{plugin}-file_chunk_size>>.
+Banding and sorting may be useful if you want some events from all files to appear in Kibana as early as possible.
+
+The plugin has two modes of operation, Tail mode and Read mode.
+
+===== Tail mode
+
+In this mode the plugin aims to track changing files and emit new content as it's
+appended to each file. In this mode, files are seen as a never ending stream of
+content and EOF has no special significance. The plugin always assumes that
+there will be more content. When files are rotated, the smaller or zero size is
+detected, the current position is reset to zero and streaming continues.
+A delimiter must be seen before the accumulated characters can be emitted as a line.
+
+===== Read mode
+
+In this mode the plugin treats each file as if it is content complete, that is,
+a finite stream of lines and now EOF is significant. A last delimiter is not
+needed because EOF means that the accumulated characters can be emitted as a line.
+Further, EOF here means that the file can be closed and put in the "unwatched"
+state - this automatically frees up space in the active window, This mode also
+makes it possible to process compressed files as they are content complete.
```

Comment:
WRT: frees up space in the active window, This mode also.
Change comma to a period